laiaaa0/assignment2/BinaryTree.cpp

```diff
@@ -22,32 +21,8 @@ BinaryTree<T>::BinaryTree(const T & value, BinaryTree<T> * left, BinaryTree<T> *
   this->right_subtree_ = right;
   this->left_subtree_->parent_ = this;
```

Comment:
Minor comment that it will be good to check that the given nodes don't have a parent yet.